### PR TITLE
Clock Stopwatch App

### PR DESCRIPTION
**Pull request by RogueMaster**
ClockApp created by CompaqDisc. Modified by me to add stopwatch features. red_link2 provided clock graphic.

# What's new

- [ Describe changes here ]

# Verification 

- [ Describe how to verify changes ]

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by hedger** [comment]
Plese don't set checkboxes in Checklist yourself, they are marked "for reviewer" for a reason.

File: applications/clock_app/clock_app.c [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+int32_t clock_app_old(void* p) {
+    UNUSED(p);
+    timerStarted=false;
+    timerSecs=0;
+    osMessageQueueId_t event_queue = osMessageQueueNew(8, sizeof(PluginEvent), NULL);
+    ClockState* plugin_state = malloc(sizeof(ClockState));
+    clock_state_init(plugin_state);
+    ValueMutex state_mutex;
+    if (!init_mutex(&state_mutex, plugin_state, sizeof(ClockState))) {
+        FURI_LOG_E(TAG, "cannot create mutex\r\n");
```
Comment (hedger) [inline applications/clock_app/clock_app.c:176]: event_queue is leaked here

File: applications/applications.c [hunk 2]
```diff
@@ -17,6 +17,7 @@ extern int32_t desktop_srv(void* p);
 extern int32_t updater_srv(void* p);
 
 // Apps
+extern int32_t clock_app_old(void *p);
```
Comment (hedger) [inline applications/applications.c]: Why is it "old", where's "new"?

File: applications/clock_app/clock_app.c [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+#include <gui/gui.h>
+#include <input/input.h>
+
+#define TAG "Clock"
+#define CLOCK_DATE_FORMAT "%.4d-%.2d-%.2d"
+#define CLOCK_TIME_FORMAT "%.2d:%.2d:%.2d"
+
+bool timerStarted=false;
+int timerSecs=0;
```
Comment (hedger) [inline applications/clock_app/clock_app.c:16]: Don't use global variables, make a state structure and allocate it while app is running.

File: applications/clock_app/clock_app.c [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    furi_hal_rtc_get_datetime(&state->datetime);
+}
+
+// Runs every 1000ms by default
+static void clock_tick(void* ctx) {
+    furi_assert(ctx);
+    osMessageQueueId_t event_queue = ctx;
+    PluginEvent event = {.type = EventTypeTick};
+    if(timerStarted) {
+        timerSecs=timerSecs+1;
```
Comment (hedger) [inline applications/clock_app/clock_app.c:88]: Guaranteed to have poor accuracy.

**Review by hedger — COMMENTED** [review]
Please format code according to style guide.

**Comment by RogueMaster** [comment]
Thanks for the feedback. Unfortunately, I'm out of my depth on how to make the needed changes. If anyone wants to take this over, you are welcome to.

File: applications/applications.c (hunk 2 above)
Comment (RogueMaster) [reply applications/applications.c]: CompaqDisc was working on a version with Settings and Alarm

**Review by jimilinuxguy — CHANGES_REQUESTED** [review]
I do not think you should remove these compiler warnings, they catch alot of bugs.

**Comment by skotopes** [comment]
This PR can not be accepted. Please read https://github.com/flipperdevices/flipperzero-firmware/blob/dev/CONTRIBUTING.md

**Comment by RogueMaster** [comment]
Sorry, I did not realize that it would take all my future changes to it and add it to this PR. My apologies.

**Pull request closed without merging**